### PR TITLE
Fix the maximization button with a fixed inner size

### PR DESCRIPTION
**Pull request by Aursen**
This pull request fix the issue [#554](https://github.com/sixtyfpsui/sixtyfps/issues/554)

### CODE REVIEW

File: sixtyfps_runtime/rendering_backends/gl/graphics_window.rs [hunk 1]
```diff
@@ -117,7 +117,7 @@ impl GraphicsWindow {
                 // Make sure that the window's inner size is in sync with the root window item's
                 // width/height.
                 self.set_geometry(s.width, s.height);
-                window_builder.with_inner_size(s)
+                window_builder.with_inner_size(s).with_resizable(false)
```
Comment (tronical) [inline sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:120]: Thanks for the patch! I'm wondering, won't this make even windows that are meant to be resizable not resizable by the user?

Comment (Aursen) [reply sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:120]: Indeed I had only tested with slide_puzzle and memory. If preferred-width or preferred-height is set, it also blocks the maximization. I'll fix it as soon as possible.

Comment (Aursen) [reply sixtyfps_runtime/rendering_backends/gl/graphics_window.rs]: I don't know if this is the best way.

**Review by Aursen — COMMENTED** [review]
I don't know if this is the best way.

File: sixtyfps_runtime/rendering_backends/gl/graphics_window.rs [hunk 2]
```diff
@@ -93,15 +93,27 @@ impl GraphicsWindow {
...
-            (window_item.title().to_string(), window_item.no_frame())
+            (
+                window_item.title().to_string(),
+                window_item.no_frame(),
+                window_item.height(),
+                window_item.width(),
+            )
         } else {
-            ("SixtyFPS Window".to_string(), false)
+            ("SixtyFPS Window".to_string(), false, 0., 0.)
```
[the remark this answers is not in the document]
Comment (Aursen) [reply sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:106]: Yes it's true, i will correct

**Comment by tronical** [comment]
Thanks! This is much more readable. I think similar logic is also needed in `apply_window_properties`. In the `if w <= 0. || h <= 0.` branch the window is resizable, otherwise it has a fixed size - this can be applied then to the `winit::Window` via `set_resizable(bool)`.

**Comment by Aursen** [comment]
I It is not possible to do this condition in `apply_window_properties because` the size is already initialized with values.

> Thanks! This is much more readable. I think similar logic is also needed in `apply_window_properties`. In the `if w <= 0. || h <= 0.` branch the window is resizable, otherwise it has a fixed size - this can be applied then to the `winit::Window` via `set_resizable(bool)`.

**Comment by tronical** [comment]
> It is not possible to do this condition in `apply_window_properties` because the size is already initialized with values.
> 
> > Thanks! This is much more readable. I think similar logic is also needed in `apply_window_properties`. In the `if w <= 0. || h <= 0.` branch the window is resizable, otherwise it has a fixed size - this can be applied then to the `winit::Window` via `set_resizable(bool)`.

Ah yes, that's true.

**Review by tronical — APPROVED** [review]

**Comment by tronical** [comment]
I think that this might not work with the live-preview when re-using a window. For example when previewing a `.60` file that has a fixed width (window becomes not resizable) and then switching to something that uses a layout. But Olivier had the idea that it might be possible to detect this by looking at the min/max and see if they differ - and that's perhaps also how this could be done in apply_window_properties.

I'll check it out separately though, we can merge this first.

**Pull request merged**